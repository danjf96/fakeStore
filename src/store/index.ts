
import { applyMiddleware, combineReducers, createStore } from 'redux'
import products from './ducks/products'
import shoppingCart from './ducks/shoppingCart'
import categoriesProduct from './ducks/categoriesProduct'
import ReduxThunk from 'redux-thunk';
import { configureStore } from '@reduxjs/toolkit';
import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

const reducer = {
    products,
    shoppingCart,
    categoriesProduct
}
const store = configureStore({
    reducer
})

export type RootState = ReturnType<typeof store.getState>
export type AppDispatch = typeof store.dispatch

// Use throughout your app instead of plain `useDispatch` and `useSelector`
export const useAppDispatch = () => useDispatch<AppDispatch>()
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector

export default store

