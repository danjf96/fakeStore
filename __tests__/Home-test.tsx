/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import { act, fireEvent, render, waitFor } from 'react-native-testing-library';
 import Home from '../src/screens/home';
 import store from '../src/store'
 import { Provider } from 'react-redux';
 
 const component = (
   <Provider store={store} > <Home navigation={{ navigate: (screen:'string') => null }}/> </Provider>
 );
 const fakeTime = (time: number) => new Promise( (resolve, reject) => setTimeout( () => resolve(null) ,time))
 
 it('renders correctly', async () => {
   const screen = render(component)
   await waitFor( () => expect(screen.toJSON()).toMatchSnapshot()) 
 });
 
 describe('test loading and render items or not', () => {
   const storeProducts = store.getState().products
 
   test('renders correctly loading', async () => {
 
     const { getByTestId } = render( component )   
     const { list, newsList} = store.getState().products
     expect(list).toHaveLength(0)
     expect(newsList).toHaveLength(0)
     expect(getByTestId('loading')).toBeTruthy()
 
   });
 
   test('renders correctly card products', async () => {
     const onPress = jest.fn()
     const onPressSecond = jest.fn()
 
     const { getAllByTestId } = await waitFor(() =>
         render(component)
     ); 
 
     await waitFor( () => {
       const { list, newsList } = store.getState().products
       expect(list).not.toHaveLength(0)
       expect(newsList).not.toHaveLength(0)
       //expect(getAllByTestId('product')).not.toHaveLength(0)
     })
 
   });
   
}) 
   