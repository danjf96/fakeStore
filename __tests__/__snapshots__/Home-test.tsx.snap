// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  " ",
  <RCTSafeAreaView
    emulateUnlessSupported={true}
  >
    <View
      style={
        Array [
          Object {
            "padding": 19,
            "width": "100%",
          },
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": true,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="buttonLeft"
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#85868A",
                  "fontFamily": "WorkSans-Regular",
                  "fontSize": 14,
                  "fontWeight": "700",
                },
                Object {
                  "color": "black",
                  "fontSize": 18,
                  "marginBottom": 20,
                },
              ]
            }
          >
            Produtos
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#85868A",
              "fontFamily": "WorkSans-Regular",
              "fontSize": 14,
              "fontWeight": "700",
            }
          }
        />
        <View
          accessible={true}
          collapsable={false}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="buttonRight"
        >
          <Image
            source={
              Object {
                "testUri": "../../../src/assets/images/logo_icon.png",
              }
            }
            style={
              Object {
                "height": 21.5,
                "resizeMode": "contain",
                "width": 18.33,
              }
            }
            testID="imageRight"
          />
        </View>
      </View>
      <ActivityIndicator
        color="#8775FE"
        size="large"
        testID="loading"
      />
    </View>
  </RCTSafeAreaView>,
  " ",
]
`;
